pkg/pool-manager/pool.go

```diff
@@ -315,6 +315,101 @@ func GetMacPoolSize(rangeStart, rangeEnd net.HardwareAddr) (int64, error) {
 	return endInt - startInt + 1, nil
 }
 
+func (p *PoolManager) createOrUpdateMacEntryInMacPoolMap(macAddress, instanceFullName, macInstanceKey string) {
+	p.macPoolMap[macAddress] = macEntry{
+		instanceName:         instanceFullName,
+		macInstanceKey:       macInstanceKey,
+		transactionTimestamp: noPendingTransaction,
+	}
+}
+
+// getInstanceMacMap creates a subset map from macPoolMap, holding only macs that belongs to a specific instance (pod/vm)
+func (p *PoolManager) getInstanceMacMap(instanceName string) (map[string]macEntry, error) {
```

Comment:
I agree. I think PoolManager should become transactionManager and it should hold macPoolMap instance.  the macPoolMap package will then hold functions like `filterInByInstsanceName`, `GetMacEntry`, `Set`, etc... 
But since this PR is big enough, is it ok if we do it in a separate refactoring PR?

Reply:
move to macMap, but not sure what you meant with changing the reciever name. do you mean that macPoolMap will be in a separate package?

Reply:
Was talking about same package but as function for macMap.